cheatsheets/JSON_Web_Encryption_Cheat_Sheet.md

```diff
@@ -0,0 +1,360 @@
+# JSON Web Encryption (JWE) Cheat Sheet
+
+## 1. Introduction
+
+JSON Web Encryption (JWE) is a standard for encrypting JSON-based tokens, ensuring the confidentiality of claims. It extends the JSON Web Token (JWT) specification by adding encryption capabilities, making it suitable for securely transmitting sensitive data.
+
+---
+
+## 2. Key Differences Between JWT and JWE
+
+| **Feature**         | **JWT (JSON Web Token)**                             | **JWE (JSON Web Encryption)**                    |
+|---------------------|------------------------------------------------------|--------------------------------------------------|
+| **Purpose**         | Tokenized authentication and integrity verification  | Secure transmission of encrypted claims          |
+| **Protection**      | Uses digital signatures for integrity (but payload is visible) | Encrypts payload to ensure confidentiality        |
+| **Example Use Case**| Identity claims in OAuth                             | Protecting sensitive user data                   |
+
+> **Note:** JWT ensures authenticity but does NOT encrypt data, while JWE provides full confidentiality and also ensures integrity through authenticated encryption (AEAD).
+
+### ***Using JWE-Only JWTs: Is It Secure?***
+
+Some applications use JWTs only with JWE (encryption) without signing them using JWS. While this approach ensures confidentiality, it introduces potential risks:
+
+**Problem:** Since there’s no digital signature (JWS), the recipient cannot verify the sender’s authenticity.
+
+### Safe Usage Scenarios:-
+
+- If the decryption key is tightly controlled and only trusted parties can encrypt messages.
```

Comment:
This part is weird.

"only trusted parties can encrypt messages" should be understood as "we are not using public key encryption", right?